okta/requestExecutor.go

```diff
@@ -295,7 +295,10 @@ func (re *RequestExecutor) doWithRetries(ctx context.Context, req *http.Request)
 	}
 	operation := func() error {
 		resp, err = re.httpClient.Do(req.WithContext(ctx))
-		if err != nil {
+		if err == io.EOF {
```

Comment:
@bogdanprodan-okta @bretterer please can you change it to `errors.Is(err, io.EOF)` and use `errors.Is` in the future? okta-golang-sdk lets users provide their own httpClient instance. This means users may choose to have a custom transport implementation which wraps the errors. E.g. they may wish to add stacktraces to every error using https://github.com/pkg/errors library. It's really not an uncommon use case. Please see this code:

```
package main

import (
	"fmt"
	"io"

	"github.com/pkg/errors"
)

func main() {
	err := errors.WithStack(io.EOF)
	fmt.Println(err == io.EOF)
	fmt.Println(errors.Is(err, io.EOF))
}
```

Output:
```
false
true
```